server/resp/src/main/java/org/infinispan/server/resp/RespCommand.java

```diff
@@ -138,27 +140,35 @@ public static RespCommand fromByteBuf(ByteBuf buf, int commandLength) {
          return null;
       }
       for (RespCommand possible : target) {
-         byte[] possibleBytes = possible.bytes;
-         if (commandLength == possibleBytes.length) {
-            boolean matches = true;
-            // Already checked first byte, so skip that one
-            for (int i = 1; i < possibleBytes.length; ++i) {
-               byte upperByte = possibleBytes[i];
-               byte targetByte = buf.getByte(readOffset + i);
-               if (upperByte == targetByte || upperByte + 32 == targetByte) {
-                  continue;
-               }
-               matches = false;
-               break;
-            }
-            if (matches) {
-               return possible;
-            }
+         if (possible.match(buf, commandLength, readOffset)) {
+            return possible;
          }
       }
       return null;
    }
 
+   public final boolean match(byte[] other) {
+      return match(Unpooled.wrappedBuffer(other), other.length, 0);
+   }
+
+   private boolean match(ByteBuf buf, int length, int offset) {
+      byte[] possibleBytes = bytes;
+      if (length == possibleBytes.length) {
+         boolean matches = true;
+         for (int i = 0; i < possibleBytes.length; ++i) {
```

Comment:
Technically this is adding an additional case insensitive char comparison, but I guess that is okay.